gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/jsr223/GremlinScriptChecker.java

```diff
@@ -34,15 +34,16 @@ public class GremlinScriptChecker {
     /**
      * An empty result whose properties return as empty.
      */
-    public static final Result EMPTY_RESULT = new Result(null, null);
+    public static final Result EMPTY_RESULT = new Result(null, null, null);
 
     /**
      * At least one of these tokens should be present somewhere in the Gremlin string for {@link #parse(String)} to
      * take any action at all.
      */
+    // todo: why not used `Tokens`?
```

Comment:
not sure what this todo was for, but you should include the `Tokens` version here too in the caps form.

Reply:
There's a bit of a mess with names for this constants. 
For example we have constant for [evaluationTimeout](https://github.com/apache/tinkerpop/blob/master/gremlin-util/src/main/java/org/apache/tinkerpop/gremlin/util/Tokens.java#L78) and can use `Token.ARGS_EVAL_TIMEOUT`, not plain string. Same for `requestId`, but not for `scriptEvaluationTimeout`. Also `Tokens.ARGS_SCRIPT_EVAL_TIMEOUT` is not defined and I don't understand why.

I think better to use constants when possible and this like can looks like
```
private static final Set<String> tokens = new HashSet<>(Arrays.asList(Tokens.ARGS_EVAL_TIMEOUT, Tokens.ARGS_SCRIPT_EVAL_TIMEOUT,
                                                 Tokens.REQUEST_ID, Tokens.ARGS_MATERIALIZE_PROPERTIES));
```
but made in the same style as before.

Reply:
We have to check all the forms. if you just did:

```
private static final Set<String> tokens = new HashSet<>(Arrays.asList(Tokens.ARGS_EVAL_TIMEOUT, Tokens.ARGS_SCRIPT_EVAL_TIMEOUT,
                                                 Tokens.REQUEST_ID, Tokens.ARGS_MATERIALIZE_PROPERTIES));
```

then if the user sent a string query of: `g.with(Tokens.ARGS_EVAL_TIMEOUT, 100)` the checker wouldn't find it. `Tokens.ARGS_EVAL_TIMEOUT` is basically a valid part of the Gremlin string someone might send.

Reply:
I suppose you could do this:

```java
private static final Set<String> tokens = new HashSet<>(Arrays.asList(Tokens.ARGS_EVAL_TIMEOUT, Tokens.ARGS_SCRIPT_EVAL_TIMEOUT,
"ARGS_EVAL_TIMEOUT", "ARGS_SCRIPT_EVAL_TIMEOUT",
                                                 Tokens.REQUEST_ID, Tokens.ARGS_MATERIALIZE_PROPERTIES
                                                 "REQUEST_ID", "ARGS_MATERIALIZE_PROPERTIES"));
```

But then the other problem is that the script checker is in `gremlin-core` and `Tokens` is in `gremlin-driver`. It's a little messed up. I suppose the right way to resolve it would be to officially make these settings part of `gremlin-language` but that seems out of scope for your work. You could create something in JIRA for that if you want to come back to it.

Reply:
added in https://github.com/apache/tinkerpop/pull/1843/commits/8c36b01b673982875d9d9e04d91f3ddce73860f5